pkg/scheduler/localCapacity.go

```diff
@@ -0,0 +1,86 @@
+package scheduler
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/the-maldridge/nbuild/pkg/source"
+)
+
+func NewLocalCapacityProvider(l hclog.Logger, path string) *LocalCapacityProvider {
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		absPath = path
+	}
+	x := LocalCapacityProvider{
+		l:       l.Named("capacityProvider"),
+		path:    absPath,
+		ongoing: nil,
+	}
+	return &x
+}
+
+// Wrapper function for pkgCmd.Run()
+func (c *LocalCapacityProvider) pkgRun(cmd *exec.Cmd) {
+	output, err := cmd.CombinedOutput()
+	c.ongoing = nil
+	if err != nil {
+		c.l.Warn("Error building pkg", "err", err)
+	}
+	c.l.Trace("Building package output", "output", string(output))
+}
+
+// Builds a package.
+func (c *LocalCapacityProvider) DispatchBuild(b Build) error {
+	if c.ongoing != nil {
+		return errors.New("no capacity")
```

Comment:
This should be a typed error.  You can either create a new type which implements the Error interface or you can make this a package variable and return it from there.  (hint: the package variable needs to be from the scheduler since all capacity providers may wish to signal they have insufficient capacity to complete the request.